app/models/school.rb

```diff
@@ -85,4 +88,18 @@ def invite_school_contact
       false
     end
   end
+
+private
+
+  def validate_can_order_against_cap
+    if can_order? && device_allocations.all? { |allocation| allocation.cap.zero? }
+      errors.add(:order_state, 'cannot change as all caps are zero')
+    end
+  end
+
+  def validate_can_order_for_specific_circumstances_against_allocation
+    if can_order_for_specific_circumstances? && device_allocations.all? { |allocation| allocation.allocation.zero? }
```

Comment:
I'm not sure if this line is correct. Can someone confirm?

Reply:
Yes i think this is right.